### DISCUSSION TITLE
How to convert between Clojure maps and Python dictionaries?

### DISCUSSION THREAD

**Discussion by wjke-nics** [basilisp-lang/basilisp#1196] [Q&A] [answered]
I'm trying to work with both Clojure maps and Python dictionaries in my Basilisp code. Could you please provide examples of:

1. Converting a Clojure persistent map to a Python dict
2. Converting a Python dict back to a Clojure persistent map

A simple code example showing both conversions would be very helpful. Thanks!

**Comment by ikappaki** [comment, marked as answer]
Hi @wjke-nics,

You can use the [lisp->python](https://basilisp.readthedocs.io/en/latest/api/core.html#basilisp.core/lisp-%3Epy) to recursively convert Basilisp data structures into Python data structures,. These includes converting keyword keys into strings by default:
```clojure
basilisp.user=> (lisp->py {:ten {:eleven 11}})
#py {"ten" #py {"eleven" 11}}
```


Use the  [py->lisp](https://basilisp.readthedocs.io/en/latest/api/core.html#basilisp.core/py-%3Elisp) to recursively convert Python data structures into Basilisp data structures. These includes converting string keys into keywords by default:
```clojure
basilisp.user=> (py->lisp #py {"twelve" #py {"thirteen" 13}})
{:twelve {:thirteen 13}}
```

In addition

You can use the `#py {...}` reader macro to define a Python Dictionary in Basilisp  (see [Data Readers](https://basilisp.readthedocs.io/en/latest/reader.html#data-readers)):
```clojure
basilisp.user=> #py {"six" 6}
#py {"six" 6}
```

Use the `python/dict` function to convert a Basilisp map into a Python dictionary:
```clojure
basilisp.user=> (python/dict {"seven" 7})
#py {"seven" 7}
```

I hope this helps!
